src/spatialdata/_core/spatialdata.py

```diff
@@ -210,6 +210,84 @@ def from_elements_dict(elements_dict: dict[str, Union[SpatialElement, AnnData]])
     def query(self) -> QueryManager:
         return self._query
 
+    def aggregate(
+        self,
+        values: DaskDataFrame | GeoDataFrame | SpatialImage | MultiscaleSpatialImage,
+        by: str,
+        region_key: str = "region",
+        instance_key: str = "instance_id",
+        **kwargs: Any,
+    ) -> SpatialData:
+        """
+        Aggregate values by given region.
+
+        Parameters
+        ----------
+        values
+            Values to aggregate.
+        by
+            Regions to aggregate by.
+        id_key
+            Key to group observations in `values` by. E.g. this could be transcript id for points.
+            Defaults to `FEATURE_KEY` for points, required for shapes.
+        value_key
+            Key to aggregate values by. This is the key in the values object.
+            If nothing is passed here, assumed to be a column of ones.
+            For points, this could be probe intensity.
+        agg_func
+            Aggregation function to apply over point values, e.g. "mean", "sum", "count".
+            Passed to :func:`pandas.DataFrame.groupby.agg` or from :func:`xrspatial.zonal_stats`
+            according to the type of `values`.
+        target_coordinate_system
+            Coordinate system to transform to before aggregating.
+        region_name
+            Name of the region `by`.
+        region_key
```

Comment:
Why do we need to pass this? Can't we read it from the table?

Reply:
not really no, cause it represent the region of the new aggregated table. So it must be a new one.

Reply:
Got it! I'll make the docstring more explicit about this.

Reply:
Done.